packages/js/product-editor/src/blocks/product-fields/shipping-class/edit.tsx

```diff
@@ -122,15 +122,19 @@ export function Edit( {
 		throw error;
 	}
 
-	const { shippingClasses } = useSelect( ( select ) => {
-		const { getProductShippingClasses } = select(
-			EXPERIMENTAL_PRODUCT_SHIPPING_CLASSES_STORE_NAME
-		);
-		return {
-			shippingClasses:
-				getProductShippingClasses< ProductShippingClass[] >() ?? [],
-		};
-	}, [] );
+	const { shippingClasses } = useSelect(
+		( select ) => {
+			const { getProductShippingClasses } = select(
+				EXPERIMENTAL_PRODUCT_SHIPPING_CLASSES_STORE_NAME
+			);
+			return {
+				shippingClasses: isInSelectedTab
+					? getProductShippingClasses< ProductShippingClass[] >()
```

Comment:
Considering the previous implementation, the `getProductShippingClasses` can return null so we should consider this case here too.

Reply:
Done in commit [a81fe0d](https://github.com/woocommerce/woocommerce/pull/47196/commits/a81fe0d18a8f43d4933339be6811842735252361)